### ISSUE TITLE
STEP/dwg files

### ISSUE THREAD

**Issue by wifasoi** [mage0r/AutoChanger#1]
Hi,
I know i can install sketchup and export from there the necessary files, but i'm a linux user and installing sketchup and use it with wine is a pain (I tried and doesn't work, unfortunately). -*github whiny/entitled user mode engaged*- Can you export the project in for an export of the files in a STEP format (preferred, otherwise stl is workable too), or in this case DWG/dxf/svg would be fine too (but i think there is more work involved). If you don't want to track those files, you can put in a github release (so you don't have to worry about out of sync files).

This is because I want to create a version that work with the ayab board (shhh, is a secret... the ayab peaple dosn't know about it... yet :P) and use a sx1508/9 (I2C IO expander with pwm).

You can close this issue if you don't have time to do it or you think I was too rude... I don't mind.
Thank you for your consideration.

**Comment by wifasoi** [comment]
-*event more entiteld github user mode engaged*-
the license requires it   https://github.com/mage0r/AutoChanger/blob/9e84eed463dcbbb184bd24fda50bc8b92860e9d6/Hardware/TAPR_Open_Hardware_License_v1.0.txt#L94

**Comment by mage0r** [comment]
Ah, also, there is a serial/software i2c 4 pin port broken out on the circuit board design that was always intended to provide an uplink to a controller.

You are of course clear to solve the problem any way you like, but having an independent controller provides a few extra advantages.   For example, the user can move threads or change order in the event of a failure.

I've been tempted to do some work with the AYAB project a few times but have been a bit time limited and focused on approaches that are controller agnostic.

**Comment by wifasoi** [comment]
I choose the Semtex because you can read key and command LEDs.. but unfortunately I can control servo (learned the hard way)... (I'll use a combination of mpc23017 and some low cost io expander).
And leave the ayab controller do the dirty work... So I don't have to program another micro.
Another reason is to minimize the cost of (ipotetical) product (don't want to transform in a real product, I like to challenge myself :D).

And for the sake of argument:
In theory you can change license whenever you want.. so the rule do not apply to the copyright holder. You are right (if you are the only copyright holder)

And for the sake of argument, again,
maibe DWG is not the best example, but the other are so main stream that are considered "exchange format" between cad programs (STL is the JPEG of the 3d models... I despise them as a exchange format). In the license there are example of "Gerber" and "pdf", once proprietary now so whidespread that are considered "exchange format".

And for last: you can make a GitHub release anytime you want(don't think you need to tag your repo), and you can add out-of-tree files (they are not added to the repo), so it's perfect for adding binary files specific to a commit (or in this case, cad export).

And thanks for your consideration.

**Comment by wifasoi** [comment]
Wow, that was log...
TL;DR: you're right. I meant any cad exchange file. GitHub release are perfect to add out-of-tree files. And thanks for your time.

**Comment by mage0r** [comment]
https://github.com/mage0r/AutoChanger/releases/tag/1.6
